pkg/gas/priorityfee.go

```diff
@@ -0,0 +1,46 @@
+package gas
+
+import (
+	"math/big"
+	"slices"
+
+	"github.com/status-im/go-wallet-sdk/pkg/ethclient"
+)
+
+// calculatePriorityFeeFromHistory calculates priority fee for a given percentile from fee history
+func calculatePriorityFeeFromHistory(feeHistory *ethclient.FeeHistory, nBlocks int, rewardsPercentileIndex int) *big.Int {
+	priorityFees := extractPriorityFeesFromHistory(feeHistory, rewardsPercentileIndex)
+
+	startIdx := max(len(priorityFees)-nBlocks, 0)
+	return calculatePriorityFee(priorityFees[startIdx:])
+}
+
+// extractPriorityFeesFromHistory extracts priority fees for a given rewardsPercentileIndex from fee history
+func extractPriorityFeesFromHistory(feeHistory *ethclient.FeeHistory, rewardsPercentileIndex int) []*big.Int {
+	priorityFees := make([]*big.Int, 0)
+	for _, blockRewards := range feeHistory.Reward {
+		if rewardsPercentileIndex < len(blockRewards) && blockRewards[rewardsPercentileIndex] != nil {
+			priorityFees = append(priorityFees, new(big.Int).Set(blockRewards[rewardsPercentileIndex]))
+		}
+	}
+
+	return priorityFees
+}
+
+func calculatePriorityFee(priorityFees []*big.Int) *big.Int {
+	if len(priorityFees) == 0 {
+		return big.NewInt(1000000000) // 1 gwei fallback
+	}
+
+	// Calculate median of the collected priority fees for stability
+	sortedFees := make([]*big.Int, len(priorityFees))
+	copy(sortedFees, priorityFees)
+	slices.SortFunc(sortedFees, func(a, b *big.Int) int {
+		return a.Cmp(b)
+	})
+
+	medianIndex := len(sortedFees) / 2
```

Comment:
For an even number of elements, it will return the second middle element instead of the average. Suggestion:
```suggestion

	n := len(sortedFees)
	if n%2 == 1 {
	    // Odd: [1,3,5] gwei -> median = 3 gwei 
		return new(big.Int).Set(sortedFees[n/2]) 
	}
	// Even: [1,2,4,5] gwei -> median = (2+4)/2 = 3 gwei
	median := new(big.Int).Add(sortedFees[n/2-1], sortedFees[n/2])
	return median.Div(median, big.NewInt(2))

```